binstar_client/utils/__init__.py

```diff
@@ -255,11 +253,16 @@ def recursive_update(d, u):
                             }
                   }
 
+
+def has_environment_variable_config():
+    return 'ANACONDA_DEFAULT_SITE' in os.environ
+
+
 def get_config(user=True, site=True, remote_site=None):
     config = DEFAULT_CONFIG.copy()
-    if site:
+    if not has_environment_variable_config() and site:
```

Comment:
@malev I don't think this is the correct solution either. In this case when `ANACONDA_DEFAULT_SITE ` is defined, the entire config file will be ignored. What if the user has set other config variables like `{"verify_ssl": "/path/to/root/ca.pem" }`